### ISSUE TITLE
Support for batch_size > 1 for inference with datasets with images of different sizes

### ISSUE THREAD

**Issue by kavyasreedhar** [open-mmlab/mmsegmentation#1370]
Following up on closed issue: https://github.com/open-mmlab/mmsegmentation/issues/125

Hi, are there any updates since this issue was closed on handling padding when datasets consists of images with different image sizes? When modifying samples_per_gpu in test.py, there is a call to torch.stack which requires the tensors to be the same size. As a result, I believe this issue occurs for these datasets regardless of single or multi gpu inference and wanted to check if there was an existing solution that supports batch_size > 1 for inference for datasets such as ADE20K with images of different sizes.

**Comment by MengzhangLI** [comment]
It is not supported yet. We do not have a clear plan for batch inference due to lack of developing resources.

**Comment by kavyasreedhar** [comment]
Got it, thanks. I would find this feature helpful, so I am happy to help get batch inference working and have been trying to find some ways forward.

Would it be valid to resize the images as done in the training pipeline in the inference pipeline as well to size the images uniformly for inference? When I tried this, the model execution seems fine, but I run into an indexing error in intersect_and_union in metrics.py (https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/core/evaluation/metrics.py#L75) since the pred_label and label are not of the same size. This makes sense because the actual true label for the image will be the true image size while the pred_label will be for the resized image size to get all the images to have the same size. Is there a way to adjust the true label to fix this issue?

Alternatively, would you recommend handling padding at the start (and un-padding at the end) in the model itself to avoid resizing the image directly and to preserve the original label and image dimensions? How would you recommend this padding be done?

Thanks for your help!
